### ISSUE TITLE
Screen does not scroll correctly when using an Accessibility Screen Reader

### ISSUE THREAD

**Issue by hamishnorton** [flutter/flutter#104036]
Internal: b/246320262

### Given

1. Flutter web application with content that is greater than the height of the view port
2. Using a Screen Reader ( [[NVDA](https://www.nvaccess.org/download/)](https://www.nvaccess.org/download/), version: 2021.3.3) to navigate
3. Browsers
    1. Google Chrome: Version 101.0.4951.67 (Official Build) (64-bit)
    2. **Microsoft Edge:** Version 101.0.1210.39 (Official build) (64-bit)
4. Flutter (3.0.0, 2.10.3)
    
    Doctor summary (to see all details, run flutter doctor -v):
    [√] Flutter (Channel stable, 3.0.0, on Microsoft Windows [Version 10.0.19043.1706], locale en-NZ)
    [√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    [√] Chrome - develop for the web
    [√] Visual Studio - develop for Windows (Visual Studio Professional 2019 16.11.5)
    [√] Android Studio (version 2021.2)
    [√] IntelliJ IDEA Ultimate Edition (version 2022.1)
    [√] VS Code (version 1.67.1)
    [√] Connected device (3 available)
    ! Device emulator-5554 is offline.
    [√] HTTP Host Availability
    
    - No issues found!
    
    Doctor summary (to see all details, run flutter doctor -v):
    [√] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.19043.1706], locale en-NZ)
    [√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    [√] Chrome - develop for the web
    [√] Visual Studio - develop for Windows (Visual Studio Professional 2019 16.11.5)
    [√] Android Studio (version 2021.2)
    [√] IntelliJ IDEA Ultimate Edition (version 2022.1)
    [√] VS Code (version 1.67.1)
    [√] Connected device (3 available)
    [√] HTTP Host Availability
    
    - No issues found!

### When

1. You run the application and adjust the browser view port to 350px
2. Then navigate  reading with NVDA by pressing “Arrow Down” to read the whole page

### Then

1. You’ll notice that the page scrolling jumps around, and doesn’t scroll with the items that are being read out.

**Expected results:** 

Page to scroll with the navigation of the screen reader, as it does for other web pages

**Actual results:** 

The page doesn't scroll, and the scrollbar jumps, but doesn't scroll down.

<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const App());
}

class App extends StatelessWidget {
  const App({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    RendererBinding.instance.setSemanticsEnabled(true);
    return MaterialApp(
      title: 'Flutter NVDA pageflap',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const Screen(),
    );
  }
}

class Screen extends StatelessWidget {
  const Screen({
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: LayoutBuilder(
        builder: (BuildContext context, BoxConstraints viewportConstraints) {
          return SingleChildScrollView(
            child: Container(
              constraints: BoxConstraints(
                minHeight: viewportConstraints.maxHeight,
                maxWidth: 300,
              ),
              padding: const EdgeInsets.all(50.0),
              child: Column(
                mainAxisAlignment: MainAxisAlignment.start,
                children: [
                  const Text(
                    'Adjust browser window height to: 350',
                  ),
                  Text(
                    'Top. current screen height: ${MediaQuery.of(context).size.height}',
                  ),
                  const Text(
                    'Start NVDA, use "Arrow Down" to read the text.',
                  ),
                  const Text(
                    'Notice how the scroll jumps',
                  ),
                  const Text(
                    'The page should scroll with the reading',
                  ),
                  const Text(
                      '1. Start. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. 1. End.'),
                  const Text(
                      '2. Start. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. 2. End'),
                  const Text(
                      '3. Start. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. 3. End'),
                  const Text(
                      '4. Start. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. 4. End.'),
                  const Text(
                    'Bottom.',
                  ),
                ],
              ),
            ),
          );
        },
      ),
    );
  }
}

```

**Comment by maheshmnj** [comment]
Hi @hamishnorton, Thanks for filing the issue. I am able to reproduce the issue on stable and master channel. When the NVDA screen reader is running pressing arrow keys (up and down) doesn't scroll the flutter web app.

_In the below video I am pressing the down arrow key to scroll up and it doesn't work. Notice that on each down arrow key press it reads the text content but doesn't scroll_

_Note: to unmute the video for audio visual output_

https://user-images.githubusercontent.com/31410839/169013100-80493aa0-cd94-4166-a33c-669f2c8cd175.mp4


This issue only occurs when nvda screen reader is running, otherwise, the scroll works fine.



<details>
<summary>flutter doctor -v (windows)</summary>

```
[√] Flutter (Channel stable, 3.0.0, on Microsoft Windows [Version
    10.0.19044.1645], locale en-IN)
    • Flutter version 3.0.0 at C:\flutter_sdk\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (6 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)    
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)  
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual
      Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)    

[√] VS Code (version 1.64.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.40.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 
      10.0.19044.1645]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 101.0.4951.54    • Edge (web)        • edge    • web-javascript • Microsoft Edge
      100.0.1185.50

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
```
[√] Flutter (Channel master, 3.1.0-0.0.pre.727, on Microsoft Windows [Version
    10.0.19044.1645], locale en-IN)
    • Flutter version 3.1.0-0.0.pre.727 at C:\flutter_sdk\master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 36f9a186bf (6 hours ago), 2022-05-15 14:34:06 -0400    
    • Engine revision 1b064fd1a3
    • Dart version 2.18.0 (build 2.18.0-116.0.dev)
    • DevTools version 2.13.1

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)    
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual
      Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)    

[√] VS Code (version 1.64.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code       
    • Flutter extension version 3.40.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 
      10.0.19044.1645]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 101.0.4951.54    • Edge (web)        • edge    • web-javascript • Microsoft Edge
      100.0.1185.50

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by ditman** [comment]
This may be a severe blocker for `customer: fun`

**Comment by yjbanov** [comment]
@ditman This issue is about NVDA, while b/239991008 is about mobile Safari. Could be caused by the same underlying issue, but could also be different.

**Comment by xster** [comment]
Changing to P2 as it's a blocking issue for customer: fun in b/239991008
